module/zfs/dmu_object.c

```diff
@@ -31,6 +31,8 @@
 #include <sys/zap.h>
 #include <sys/zfeature.h>
 
+int zfs_metadnode_backfill = 0;
```

Comment:
It probably makes sense to add a pool parameter for this instead of just a module option, so that it can be set persistently if users care more about dense allocations than create performance.